### ISSUE TITLE
upgrade to latest version fails and app is removed

### ISSUE THREAD

**Issue by niko2** [YunoHost-Apps/nextcloud_ynh#675]
### Describe the bug

Running nextcloud 28.0.2~ynh1 with custom data_dir
Upgrade to 28.0.3~ynh1 starts by moving data from custom data_dir to default deta_dir
Then fails and leave nextcloud uninstalled
The step that failed was restauration because data has been moved during upgrade (see logs below)

Several actions should not occured, so not sure where is the bug 

### Context

- Hardware: VM
- YunoHost version: 11.2.10.3 (latest)
- I have access to my server: Through SSH + through the webadmin + direct access via console
- Are you in a special context or did you perform some particular tweaking on your YunoHost instance?: yes
  - nextcloud custom data_dir
- If upgrading, current package version: *can be found in the admin, or with `yunohost app info $app_id`*
 - nextcloud has been removed during upgrade
 
### Steps to reproduce

Upgrade nextcloud from 28.0.2~ynh1 to 28.0.3~ynh1 with custom data_dir

### Expected behavior

- Don't move data to defaut data_dir
- Eventually fails but leave nextcloud intact in version 28.0.2~ynh1

### Logs

Backup creation (OK): https://paste.yunohost.org/raw/wepetegacu
Upgrade (KO): https://paste.yunohost.org/raw/iwimagifig
Restore (KO): https://paste.yunohost.org/raw/atiyetubih
Remove app (OK): https://paste.yunohost.org/raw/etadacicej

**Comment by alexAubin** [comment]
Can you elaborate how you "customized" the data dir ? As far as I know all methods apart from using bind mounts will lead to this kind of stuff and there's not much we can do to fix it as this is kind of the excepted behavior

**Comment by niko2** [comment]
I understand this is not supported (by install or by upgrade scripts)
I followed the doc mentionned in this PR https://github.com/YunoHost/doc/pull/2415
TLDR; I installed nextcloud, I moved data to the new directory and I updated configuration using `yunohost app setting nextcloud data_dir -v /new/data/path`

Since it is not supported, should the upgrade script move the data or just fail ?

The upgrade process broke my system.
I guess I need to install again and restore the backup; I hope it will work

**Comment by zamentur** [comment]
Seems related to https://github.com/YunoHost-Apps/nextcloud_ynh/issues/485
